### PR TITLE
[ADVAPP-1003]: Enhance the abilities of global search by including preferred name in search results.

### PR DESCRIPTION
**Pull request by kandarp-canyon**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/ADVAPP-1003

### Technical Description

>Enhance the abilities of global search by including preferred name in search results.

### Any deployment steps required?

> No.

### Are any Feature Flags and/or Data Migrations that can eventually be removed Added?

> No.

_______________________________________________

### CODE REVIEW

File: app-modules/prospect/src/Filament/Resources/ProspectResource.php [hunk 1]
```diff
@@ -114,6 +114,7 @@ public static function getGlobalSearchResultDetails(Model $record): array
             'Other ID' => $record->otherid,
             'Email Address' => collect([$record->email, $record->email_id])->filter()->implode(', '),
             'Phone' => collect([$record->mobile, $record->phone])->filter()->implode(', '),
+            'Preferred Name' => $record->preferred,
```
Comment (Orrison) [inline app-modules/prospect/src/Filament/Resources/ProspectResource.php:117]: You also need to add `preferred` to `getGloballySearchableAttributes`.

Also, @danharrin should we include it in the `scoreGlobalSearchResults` in `modifyGlobalSearchQuery` in any way as well?

File: app-modules/student-data-model/src/Filament/Resources/StudentResource.php [hunk 2]
```diff
@@ -78,6 +78,7 @@ public static function getGlobalSearchResultDetails(Model $record): array
             'Email Address' => collect([$record->email, $record->email_id])->filter()->implode(', '),
             'Mobile' => $record->mobile,
             'Phone' => $record->phone,
+            'Preferred Name' => $record->preferred,
```
Comment (Orrison) [inline app-modules/student-data-model/src/Filament/Resources/StudentResource.php:81]: You also need to add `preferred` to `getGloballySearchableAttributes`.

Also, @danharrin should we include it in the `scoreGlobalSearchResults` in `modifyGlobalSearchQuery` in any way as well?

**Review by Orrison — CHANGES_REQUESTED** [review]

File: app-modules/prospect/src/Filament/Resources/ProspectResource.php (hunk 1 above)
Comment (kandarp-canyon) [reply app-modules/prospect/src/Filament/Resources/ProspectResource.php:117]: Fixed

File: app-modules/student-data-model/src/Filament/Resources/StudentResource.php (hunk 2 above)
Comment (kandarp-canyon) [reply app-modules/student-data-model/src/Filament/Resources/StudentResource.php:81]: Fixed

Comment (danharrin) [reply app-modules/student-data-model/src/Filament/Resources/StudentResource.php:81]: As above

File: app-modules/prospect/src/Filament/Resources/ProspectResource.php (hunk 1 above)
Comment (danharrin) [reply app-modules/prospect/src/Filament/Resources/ProspectResource.php:117]: No I don't think we need to score by this attribute for now

**Review by Orrison — APPROVED** [review]

**Pull request merged**